test/androidTest/java/org/thoughtcrime/securesms/espresso/ConversationsHelper.java

```diff
@@ -31,50 +69,14 @@ public ConversationHelper goConversation() {
 
     public ConversationHelper goGroup() {
         try {
-            this.assertText(groupName);
-            onView(withText(groupName))
-                .perform(click());
+            onView(first(anyOf(
+                withText(Helper.groupName),
+                withText("null")
+            ))).perform(click());
         } catch (NoMatchingViewException e) {
-            newGroup();
-        } finally {
```

Comment:
I see you removed "FINALLY"?

Reply:
can you clarify the reason for removing finally here so that we keep it for our records

Reply:
IntelliJ warning: `'finally' block cannot complete normally`

it says that because having that a return in finally makes it override actual return (outside of try block) or exceptions being thrown.

in this case, when the `newGroup` method was failing, it would still return as if there was no problem and fail later on in the test code. this hides the actual error and makes it super confusing.

Reply:
good justification